pkg/authz/permissions.go

```diff
@@ -79,3 +93,93 @@ func checkAccess(ctx context.Context, c *kubernetes.Clientset, np NamespacedPerm
 
 	return r.Status.Allowed, nil
 }
+
+func creatUnwantedPermissionsList(ctx context.Context, cfg *rest.Config, unwantedPermissions []string, ns string, definedRules map[[4]string]int) ([]string, error) {
+	c, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return []string{}, fmt.Errorf("error creating the client: %w", err)
+	}
+
+	ssrr := authzv1.SelfSubjectRulesReview{
+		ObjectMeta: metav1.ObjectMeta{Namespace: ns},
+		Spec:       authzv1.SelfSubjectRulesReviewSpec{Namespace: ns},
+	}
+
+	r, err := c.AuthorizationV1().SelfSubjectRulesReviews().Create(ctx, &ssrr, metav1.CreateOptions{})
+	if err != nil {
+		return []string{}, err
+	}
+	resourceRules := r.Status.ResourceRules
+	for _, rr := range resourceRules {
```

Comment:
can we reduce the number of nested `for` loops?

Reply:
I have created a new function with unit tests.